hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -504,6 +516,24 @@ Map<String, Object> getFieldSplit(List<String> fields, Map<String, Object> field
       return fieldMap;
     }
 
+    Map<String, Object> getFilterSplit(List<String> fields, Map<String, Object> fieldMap, Filter value) {
+      int len = fields.size();
+      if (fieldMap == null) {
+        fieldMap = new HashMap<>();
+      }
+      if (len == 1) {
+        fieldMap.putIfAbsent(fields.get(0), value);
+      } else {
+        Map<String, Object> fieldMapGet = (Map<String, Object>) fieldMap.get(fields.get(0));
+        if (fieldMapGet == null) {
+          fieldMap.put(fields.get(0), getFilterSplit(fields.subList(1, len), null, value));
```

Comment:
As code understanding, it may cause overwrite if prefix field is same for 2 different full key path, eg:
a.b.c.d:equals:10,a.b.m.n:equal:40
Also, in other patterns, this may have some overwrite of other filters.

Reply:
I have tested, it won't cause override unless the fields passed are exactly the same. The logic is such that if an entry exists in the map, then it will navigate into that value and add new entry. 
example: 
```
first time, 
    the function is called for (fields = a.b.c.d, fieldMap = null) 
    it will create a map like : `{a={b={c={d=EQUALS, 10}}}}`
next, 
    the function is called for (fields = a.b.m.n, fieldMap = {a={b={c={d=EQUALS, 10}}}}),  
    for "a" it will navigate the existing map, and recursively call the method with parameters(fields = b.m.n, fieldMap = {b={c={d=EQUALS, 10}}}) ---line 531
        then again for "b", recursively, the call is made for (fields = m.n, fieldMap= {c={d=EQUALS, 10}}) ---line 531
        now as the field "m" doesn't match any key in the map, put operation is performed for key=m ---line 529
        ...so on
```

Also the above is tested with a test code for just the map creation logic, 
input: 
String x = "acls.name.first:equals:NAME,acls.name.second:equals:TYPE";
output:
Map at iteration-1: {acls={name={first=EQUALS, NAME}}}
Map at iteration-2: {acls={name={first=EQUALS, NAME, second=EQUALS, TYPE}}}

input: 
String y = "a.b.c.d:equals:10,a.b.m.n:equals:40";
output:
Map at iteration-1: {a={b={c={d=EQUALS, 10}}}}
Map at iteration-2: {a={b={c={d=EQUALS, 10}, m={n=EQUALS, 40}}}}